internal/http/services/owncloud/ocs/handlers/cloud/users/users.go

```diff
@@ -69,19 +72,24 @@ func (h *Handler) GetGroups(w http.ResponseWriter, r *http.Request) {
 
 // Quota holds quota information
 type Quota struct {
-	Free       int64   `json:"free" xml:"free"`
-	Used       int64   `json:"used" xml:"used"`
-	Total      int64   `json:"total" xml:"total"`
-	Relative   float32 `json:"relative" xml:"relative"`
-	Definition string  `json:"definition" xml:"definition"`
+	Free       int64   `json:"free,omitempty" xml:"free,omitempty"`
+	Used       int64   `json:"used,omitempty" xml:"used,omitempty"`
+	Total      int64   `json:"total,omitempty" xml:"total,omitempty"`
+	Relative   float32 `json:"relative,omitempty" xml:"relative,omitempty"`
+	Definition string  `json:"definition,omitempty" xml:"definition,omitempty"`
 }
 
-// Users holds users data
-type Users struct {
-	Quota       *Quota `json:"quota" xml:"quota"`
-	Email       string `json:"email" xml:"email"`
-	DisplayName string `json:"displayname" xml:"displayname"`
-	UserType    string `json:"user-type" xml:"user-type"`
+// User holds user data
+type User struct {
+	Enabled           string `json:"enabled" xml:"enabled"`
+	UserType          string `json:"user-type" xml:"user-type"`
+	UserID            string `json:"id" xml:"id"` // UserID is mapped to the preferred_name attribute in accounts
+	DisplayName       string `json:"display-name" xml:"display-name"`
+	LegacyDisplayName string `json:"displayname" xml:"displayname"`
+	Email             string `json:"email" xml:"email"`
+	Quota             *Quota `json:"quota,omitempty" xml:"quota,omitempty"`
+	UIDNumber         int64  `json:"uidnumber" xml:"uidnumber"`
+	GIDNumber         int64  `json:"gidnumber" xml:"gidnumber"`
```

Comment:
I think `UIDNumber` and `GIDNumber` should also be optional here. Currently the CS3 API returns them always (with some hardcoded defaults) but in the long run we should change that IMO.